src/session.ts

```diff
@@ -0,0 +1,405 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import { GraphQLClient } from 'graphql-request';
+import { generateHash, KeyPair } from 'p2panda-js';
+
+import { createOperation, deleteOperation, updateOperation } from './operation';
+import { nextArgs, publish } from './graphql';
+
+import type {
+  DocumentViewId,
+  EncodedEntry,
+  EncodedOperation,
+  Fields,
+  NextArgs,
+  PublicKey,
+  SchemaId,
+} from './types';
+
+/**
+ * Options we can pass in into methods which will override the globally set
+ * options for that session for that method call.
+ */
+export type Options = {
+  /**
+   * Key pair which is used for that method call to sign the entry.
+   */
+  keyPair: KeyPair;
+
+  /**
+   * Schema Id which is used for that method call to identify the schema of the
+   * document.
+   */
+  schemaId: SchemaId;
+};
+
+/**
+ * Communicate with the p2panda network through a `Session` instance.
+ *
+ * `Session` provides a high-level interface to create data in the p2panda
+ * network by creating, updating and deleting documents following data schemas.
+ * It also provides a low-level API for directly accessing and creating
+ * entries on the Bamboo append-only log structure.
+ *
```

Comment:
What does "accessing" mean in this case? Is that retrieving next args or is the meaning elsewhere?

Reply:
I remove it as it's somehow implied by "Provides a low-level API"